core/src/main/java/com/epam/pipeline/entity/user/RunnerSid.java

```diff
@@ -17,27 +17,75 @@
 package com.epam.pipeline.entity.user;
 
 import com.epam.pipeline.entity.pipeline.run.parameter.RunAccessType;
-import lombok.AllArgsConstructor;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import lombok.Builder;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
 
 import javax.persistence.Embeddable;
 import javax.persistence.EnumType;
 import javax.persistence.Enumerated;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
 @Embeddable
 @Getter
 @Setter
 @NoArgsConstructor
 @Builder
-@AllArgsConstructor
 public class RunnerSid {
 
     private String name;
     private boolean principal = true;
 
     @Enumerated(EnumType.STRING)
     private RunAccessType accessType = RunAccessType.ENDPOINT;
+    private Boolean pipelinesAllowed;
+    private Boolean toolsAllowed;
+    @JsonIgnore
+    private String pipelinesList;
+    @JsonIgnore
+    private String toolsList;
```

Comment:
As long as we return instances of this class in `UserController::getRunners`, we shall remove `@JsonIgnore` annotations here. Otherwise, a client won't be able to read runners details once they are saved.

Other option would be to return `RunnerSidVO` instances in `UserController::getRunners`. In this case both structures would be the same.

Reply:
Client will receive parsed ids information via getters `getPipelines` and `getTools()` defined in this class. In order to switch completely from entity `RunnerSid` to `RunnerSidVO` completely we also need to add changes to `PipelineUser` class, which I prefer to avoid.